.github/workflows/fips.yml

```diff
@@ -0,0 +1,332 @@
+name: Go FIPS
```

Comment:
If we could expose a single `FIPS=0/1` environment variable or something like that then it seems like we could use the existing workflow files and use a `strategy.matrix` to do everything for FIPS and non-FIPS.

Reply:
+1. 

@keithmattix is that feasible?

Reply:
I don't think GitHub actions has the capability to conditionally set the image and other config that's necessary for the FIPS version of the build. That's the only limitation I can think of for the build matrix plan